src/Scripting/Core/Script.cs

```diff
@@ -87,10 +95,26 @@ internal static Script<T> CreateInitialScript<T>(ScriptCompiler compiler, string
             ContinueWith<object>(code, options);
 
         /// <summary>
+        /// Continues the script with given <see cref="Stream"/> representing code.
+        /// </summary>
+        /// <exception cref="ArgumentNullException">Stream is null.</exception>
+        /// <exception cref="ArgumentException">Stream is not readable or seekable.</exception>
+        public Script<object> ContinueWith(Stream code, ScriptOptions options = null) =>
+            ContinueWith<object>(code, options);
+
+        /// <summary>
         /// Continues the script with given code snippet.
         /// </summary>
         public Script<TResult> ContinueWith<TResult>(string code, ScriptOptions options = null) =>
-            new Script<TResult>(Compiler, Builder, code ?? "", options ?? InheritOptions(Options), GlobalsType, this);
+            new Script<TResult>(Compiler, Builder, SourceText.From(code ?? "", options?.FileEncoding ?? Options.FileEncoding), options ?? InheritOptions(Options), GlobalsType, this);
```

Comment:
>options?.FileEncoding [](start = 79, length = 21)

I don't think we should use ?? here -- if the FileEncoding is set to null on the current options, we shouldn't use the previous script encoding. 

Let's simplify this complex expression a bit to something like:
```
{
options = options ?? InheritOptions(Options);
return new Script(..., options.FileEncoding), options, ...)
}
```